pkg/reconciler/broker/controller.go

```diff
@@ -49,7 +50,16 @@ const (
 	controllerAgentName = "broker-controller"
 )
 
-func NewController(ctx context.Context, cmw configmap.Watcher) *controller.Impl {
+type Constructor injection.ControllerConstructor
+
+// NewConstructor creates a constructor to make a Broker controller.
+func NewConstructor(dataresidencyss *dataresidency.StoreSingleton) Constructor {
+	return func(ctx context.Context, cmw configmap.Watcher) *controller.Impl {
+		return newController(ctx, cmw, dataresidencyss.Store(ctx, cmw))
+	}
+}
+
+func newController(ctx context.Context, cmw configmap.Watcher, dataresidencyss *dataresidency.Store) *controller.Impl {
```

Comment:
```suggestion
func newController(ctx context.Context, cmw configmap.Watcher, drs *dataresidency.Store) *controller.Impl {
```

Reply:
Good catch, I was planning to change the name last Friday but somehow it skipped my mind. Thanks.